### ISSUE TITLE
Haskell stack support

### ISSUE THREAD

**Issue by ynishi** [actions/starter-workflows#141]
I tried to use haskell template, but my project was stack and failed.
How about support build with stack when to be failed cabal install.
Sample below uses if for each cabal steps(but sent mail as failed.)
```
    steps:
    - uses: actions/checkout@v1
    - name: Install dependencies
      run: cabal install --only-dependencies --enable-tests    
    - name: Install stack
      if: failure()
      run: |
        curl -sSL https://get.haskellstack.org/ | sh
        stack setup
    - name: Build
      run: |
        cabal configure --enable-tests
        cabal build
    - name: Build with stack
      if: failure()
      run: stack build
    - name: Run tests
      run: cabal test
    - name: Run tests with stack
      if: failure()
      run: stack test
```

**Comment by peaceiris** [comment]
FYI. Haskell setup action https://github.com/actions/setup-haskell

**Comment by ynishi** [comment]
Thanks, it looks good for support matrix and so on. Sample integrate,
```
    steps:
    - uses: actions/checkout@v1
    - name: Setup Haskell
    - uses: actions/setup-haskell@v1
      with:
        ghc-version: '8.6.5'
        cabal-version: '3.0'
    - name: Install dependencies
      run: cabal install --only-dependencies --enable-tests
    - name: Install stack
      if: failure()
      run: |
        curl -sSL https://get.haskellstack.org/ | sh
        stack setup
...
```

**Comment by andymckay** [comment]
Thanks @ynishi, `setup-haskell` was merged a couple of days ago in #160 so that should help you out here. If not, please let us know.
